phoenix-core/src/main/java/org/apache/phoenix/mapreduce/util/PhoenixConfigurationUtil.java

```diff
@@ -154,7 +154,8 @@
 
     public enum SchemaType {
         TABLE,
-        QUERY;
+        QUERY,
+        UPDATE_STATS
```

Comment:
I am not very familiar with the TABLE/QUERY enum, whats the difference?

Reply:
I am also not quite sure about the exact use case. The SchemaType.TABLE enum value is only used by phoenix-hive module. The other option is SchemaType.QUERY which is default option to run regular queries.

Reply:
I checked the code. THE QUERY is generally used for phoenix map reduce jobs. I doubt that we should reuse SchemaType and add UPDATE_STATS here, because now for every place where it checks whether schema type is QUERY you might need to check if it can be applied to UPDATE_STATS too, as UPDATE_STATS is also select statement and MR job. I prefer to add one more new property in configuration to indicate whether this is UPDATE STATS or not instead of adding a new UPDATE_STATS enum type to SchemaType.

Reply:
I feel that adding a new `SchemaType` is cleaner way to handle this. I agree that everywhere we check for the `SchemaType`, we need to do for this one as well, but we don't have it in many places and it provides an insight to the user about the type of MR job it is. If we add aggregation support in future, this can also be extended similarly. The name `SchemaType` is not really appropriate name here, however I didn't change it since it is public enum. It should be something like `PhoenixMRJobType`.

Adding a config property would not eliminate the checks that we have through the code and I feel that it would make it harder to manipulate in future if new stuff is added. @BinShi-SecularBird Thoughts?

Reply:
My point is that the meanings of QUERY and UPDATE_STATS have overlap -- QUERY indicates Phoenix MR jobs and UPDATE_STAS indicates Phoenix MR jobs plus update statistics, which will bring unclean code to the code base. Thoughts?

Reply:
> My point is that the meanings of QUERY and UPDATE_STATS have overlap

They have some overlap. All of them are Phoenix MR job types. However Phoenix MR is generic framework.
`QUERY` suggests `Phoenix MR jobs` that allow you to run `arbitrary queries` (without aggregations)
`UPDATE_STATS` suggests `Phoenix MR jobs` that will `collect statistics` (internally it might run a fixed query)

Depending on use case, the appropriate mappers/record readers/other MR components etc will be called.

Reply:
If you check the usages of SchemaType in the whole code base, you'll find SchemaType is mainly used by Phoenix-Pig and has its specific meaning in Phoenix-Pig. TABLE and QUERY maps to two PhoenixHBaseLoader scheme "hbase://table/" and "hbase://query", and it also maps to corresponding resource schema in Phoenix-Pig. That's why it seems to a big idea to define new schema type and pollute its usage in Phoenix_Pig. I prefer to add a new configuration or new Parameter in PhoenixMapReduceUtil.setInput to indicate this is UPDATE_STATISTICS MR job.

Reply:
@BinShi-SecularBird  Although its _primarily_ being used for phoenix-pig module, phoenix-core also uses it to determine the query it needs to run for MR job. I want to _extend_ the meaning of `SchemaType` enum to suggest this as well since that can be extended again if we have new types/use cases in future. Adding a configuration each time doesn't seem good to me.

@dbwong?

Reply:
This is my last concern about your change. I object to reuse SchemaType for indicating UPDATE_STATISTICS MR job, because we need to keep logical independence of data structure and data type. I don't want to block your progress on your work. If you insist on making this change, you can ignore my comment and proceed with commit.

Reply:
As discussed offline, we will add a new enum named `PhoenixMRJobType` to indicate that it is a stats collection job.

Reply:
@BinShi-SecularBird Updated the patch with the `MRJobType` enum.